api/src/main/java/com/datastrato/gravitino/authorization/SecurableObjects.java

```diff
@@ -115,52 +114,25 @@ public static SecurableObject ofAllMetalakes(List<Privilege> privileges) {
     return new SecurableObjectImpl(null, "*", SecurableObject.Type.METALAKE, privileges);
   }
 
-  private static class SecurableObjectImpl implements SecurableObject {
+  private static class SecurableObjectImpl extends MetadataObjects.MetadataObjectImpl
+      implements SecurableObject {
 
-    private final String parent;
-    private final String name;
-    private final Type type;
     private List<Privilege> privileges;
 
     SecurableObjectImpl(String parent, String name, Type type, List<Privilege> privileges) {
-      this.parent = parent;
-      this.name = name;
-      this.type = type;
+      super(parent, name, type);
       this.privileges = ImmutableList.copyOf(privileges);
     }
 
-    @Override
-    public String parent() {
-      return parent;
-    }
-
-    @Override
-    public String name() {
-      return name;
-    }
-
-    @Override
-    public String fullName() {
-      if (parent != null) {
-        return parent + "." + name;
-      } else {
-        return name;
-      }
-    }
-
-    @Override
-    public Type type() {
-      return type;
-    }
-
     @Override
     public List<Privilege> privileges() {
       return privileges;
     }
 
     @Override
     public int hashCode() {
-      return Objects.hash(parent, name, type, privileges);
+      int result = super.hashCode();
+      return 31 * result + Objects.hash(privileges);
```

Comment:
How about?
```
Objects.hash(super.hashCode(), privileges)
```

Reply:
It might be similar, I will change to it.